### ISSUE TITLE
Provide pre-configured Camel K instance with Camel K stack

### ISSUE THREAD

**Issue by apupier** [eclipse-che/che#15391]
### Is your enhancement related to a problem? Please describe.

currently when using a Camel K stack, user need to have an already installed Camel instance somewhere or have enough rights to install it.

### Describe the solution you'd like

when creating the workspace, a Camel K instance is available on whatever Kubernetes cluster it is installed on.
It may require an action requested to an admin to install the Camel K Operator on the cluster.

### Describe alternatives you've considered

when creating the workspace, a Camel K instance is available on whatever Kubernetes cluster it is installed on but it requires admin right if operator not available on the cluster.

As a first iteration, we can also imagine to have listed the requirements (installed operator) in the devfile and that Che is alerting user in case some of the operator is not available.

### Additional context

might be done using Operator Lifecycle Manager if supported in devfile

follow-up on https://github.com/eclipse/che/issues/14831

**Comment by l0rd** [comment]
@apupier are you requesting the support for operators as a new component type in the devfile? 

I believe that's something that we agreed last week we won't be able to support because it requires more privileges then we can grant to a Che user (privilege escalation).

**Comment by apupier** [comment]
we said that it won't be possible in short-term but that we might rely on Operator Lifecycle Manager in the future. It might need to request some rights to an admin but it should be integrated in a Kubernetes flow.
And I wasn't precise enough in the "describe the solution you'd like", i will update

**Comment by l0rd** [comment]
yeah we agreed that we would like to support Kubernetes Custom Resources, but CRD need to be deployed beforehand.

**Comment by apupier** [comment]
it stays something interesting to have in the future
